finrl/env/env_stocktrading_cashpenalty.py

```diff
@@ -247,27 +261,36 @@ def step(self, actions):
             self.account_information["reward"].append(reward)
 
             # multiply action values by our scalar multiplier and save
-            actions = actions * self.hmax
+            actions = actions * self.hmax 
+            #buy/sell only if the price is > 0 (no missing data in this particular date 
+            actions = np.array([actions[i] if closings[i]>0 else 0 for i in range(len(actions))])
+
+            if self.turbulence_threshold is not None:
+                # if turbulence goes over threshold, just clear out all positions
+                if self.turbulence>=self.turbulence_threshold:
+                    actions = -(np.array(holdings) * closings)
+                    self.log_step(reason="TURBULENCE")
             self.actions_memory.append(actions)
 
-            # scale cash purchases to asset # changes
-            actions = actions / closings
+            # scale cash purchases to asset
+            actions = actions // closings
+            #convert into integer because we can't buy fraction of shares
+            actions = (actions.astype(int))
```

Comment:
@louisowen6 I'd like this discretization of actions to be optional (i.e. )
`if self.discrete:-> do discretization of actions'`

Reply:
full disclosure, I plan to trade with this tool, but trade on an exchange that has fractional shares  (like robinhood)

Reply:
Sure! Never thought we can trade in fractional shares. This is new knowledge for me :)